### DISCUSSION TITLE
ko/patterns/template-method/

### DISCUSSION THREAD

**Discussion by giscus** [iluwatar/java-design-patterns-vuepress-web#68] [Announcements]
# ko/patterns/template-method/

의도 동작 상의 알고리즘의 뼈대를 정의하고 일부 단계들을 하위 클래스에게 미룹니다. 템플릿 메소드는 알고리즘 구조의 변경없이 하위 클래스들이 알고리즘의 특정 단계들을 재정의할 수 있게 합니다. 설명 실제 예제 일반적으로 물건을 훔치는 절차는 동일합니다. 첫번째 단계로 대상을 고릅니다. 다음 단계로 대상을 어느정도 혼란스럽게 만듭니다. 마지막 단계로 물건을 훔칩니다. 하지만, 이 단계를 실행하는 구체적인 방법에는 여러가지가 있습니다.

https://java-design-patterns.com/ko/patterns/template-method/

**Comment by cheolSS** [comment]
Hafling Thief가 반쪽자리 도적이 아니라 종족명 하플링을 말하는 것은 아닌가요? ㅋㅋ
